nemo_text_processing/g2p/modules.py

```diff
@@ -348,36 +369,82 @@ def __init__(
         )
 
         self.ignore_ambiguous_words = ignore_ambiguous_words
-        self.heteronyms = (
-            set(self._parse_file_by_lines(heteronyms))
-            if isinstance(heteronyms, str) or isinstance(heteronyms, pathlib.Path)
-            else heteronyms
-        )
-        if set_graphemes_upper and heteronyms:
-            self.heteronyms = [het.upper() for het in self.heteronyms]
+        if isinstance(heteronyms, str) or isinstance(heteronyms, pathlib.Path):
+            self.heteronyms = set(self._parse_file_by_lines(heteronyms))
+        elif isinstance(heteronyms, list) and all(isinstance(het, str) for het in heteronyms):
+            self.heteronyms = set(heteronyms)
+        else:
+            self.heteronyms = None
```

Comment:
> if they've passed something in that's not being used

This will not happen because we had type hints `heteronyms: Optional[Union[str, pathlib.Path, List[str]]] = None`, which only consider str, pathlib.Path, List, or None.